daras_ai_v2/language_model.py

```diff
@@ -1371,6 +1374,12 @@ def run_openai_chat(
             for model_id in model_ids
         ],
     )
+
+    if needs_openai_role_workaround:
```

Comment:
can you create a copy of the messages and then pass to the API instead of having to do this revert back